api/extension-guides/language-model-tutorial.md

```diff
@@ -9,13 +9,324 @@ MetaDescription: Tutorial that walks you through creating a VS Code extension th
+      {
+        "when": "editorFocus",
+        "command": "code-tutor.annotate",
+        "group": "navigation"
+      }
+    ]
+  }
+}
+```
+
+- The "editor/title" specifies where we want this menu option - which is in the title bar of whichever tab is currently open in the editor area.
+- The "when" specifies when we want this menu option to be shown. In this case, we want it to be shown whenever the editor has focus.
+- The "command" specifies which command we want to run when the user clicks this menu option. In this case, we want to run the "code-tutor.annotate" command that we defined earlier.
+- The "group" specifies where we want this menu option to be shown in the title bar. The "navigation" group is the section that is on the top right of the editor.
+
+<!-- TODO: Add "Group" screenshot -->
+![]()
+
+At this point, you can run the extension from the "Run and Debug" view in the sidebar. This will open a new VS Code instance with the extension installed. If you open a project that contains code files and then open a code file, you should see the "Tutor" button in the title bar of the editor. You'll also see a "Tutor" option in the Command Palette.
+
+However, if you click on the "Tutor" button or select the option from the Command Palette, you'll see an error message that says "Command 'code-tutor.annotate' resulted in an error (command 'code-tutor.annotate' not found)". That's because we haven't implemented the command yet, so let's do that next.
+
+## Step 3: Implement the "Tutor" command
+
+Commands are defined in the `package.json` file, but they are implemented in the `src/extension.ts` file. Open this file and locate the following line.
+
+```ts
+const disposable = vscode.commands.registerCommand('code-tutor.helloWorld', () => {
+```
+
+Remember that we changed this command in the `package.json` file to be `code-tutor.annotate` instead of `code-tutor.helloWorld`. We also want to register a TextEditor command instead of just a command. This will give us a reference to the current text editor.
+
+Modify the line so that it looks like this:
+
+```ts
+const disposable = vscode.commands.registerTextEditorCommand('code-tutor.annotate', async (textEditor: vscode.TextEditor) => {
+```
+
+You can now run the extension again, and this time when you click on the "Tutor" button or select the option from the Command Palette, you should see a message that says "Hello from Code Tutor!".
+
+We're now ready to start building the functionality to generate AI-powered code annotations.
+
+## Step 4
+
+To get our Code Tutor annotations working, we'll need to do a few things...
+
+1. Get the code with line numbers from the current editor.
+2. Send the code to the Language Model API to get the annotations.
+3. Parse the annotations and display them in the editor.
+
+Let's start by getting the code from the current editor. Add the following method directly above the `export function deactivate() { }` line at the bottom of the `extension.ts` file.
```

Comment:
Maybe include a short explanation that we'll provide the code as context for our LM request? This makes it clear that the extension decides which context is provided and that it's not something automatic.
A possible addition (as homework :) ) would be to take into account the user's selection.